modules/UI/authentication/AuthHandler.js

```diff
@@ -27,38 +28,60 @@ const getTokenAuthUrl
  * @param {string} [lockPassword] password to use if the conference is locked
  */
 function doExternalAuth(room, lockPassword) {
+    isForcePopupExternalAuth = config.enablePopupExternalAuth === true;
+
     if (externalAuthWindow) {
         externalAuthWindow.focus();
 
         return;
     }
+
     if (room.isJoined()) {
         let getUrl;
 
         if (isTokenAuthEnabled) {
             getUrl = Promise.resolve(getTokenAuthUrl(room.getName(), true));
             initJWTTokenListener(room);
         } else {
-            getUrl = room.getExternalAuthUrl(true);
+            getUrl = room.getExternalAuthUrl(isForcePopupExternalAuth);
+        }
+        if (isForcePopupExternalAuth) {
+            getUrl.then(url => {
+                openPopupAuthDialog(url, room, lockPassword);
+            });
+        } else {
+            getUrl(UIUtil.redirect);
```

Comment:
I haven't seen this syntax before--a promise instance is getting executed. That does the same thing as a .then?